### PR TITLE
Implement Private network type for Private Gateways

### PR DESCRIPTION
**Pull request by eliasgomes**
TL;DR Implement a Private network type to be used for VPC peering. 

**API changes:**

*createPrivateGateway, listPrivateGateways:*
- Removed parameters:
  - gateway
  - netmask
  - vlan / broadcast_uri
  - networkoffferingid
  - physicalnetworkid
- New required parameter (when creating):
  - networkid

*createStaticRoute:*
- Removed parameters:
  - gatewayid
- New required parameter:
  - vpcid
  - nexthop

*listStaticRoutes:*
- New in response:
  - networkid
  - networkname
  - cidr

Add a private network like this:
screen shot 2017-01-12 at 17 03 29

Per VPC, add a gateway to this network:
screen shot 2017-01-12 at 17 05 08

Gateways connected:
screen shot 2017-01-12 at 17 14 08

screen shot 2017-01-12 at 17 05 31

Networks:
screen shot 2017-01-12 at 17 07 59

You could already set the desired static routes to get traffic flowing.

Inter-VPC connectivity working:
inter-vpc-working

These are the VMs in the tiers:
screen shot 2017-01-12 at 17 07 41

### CODE REVIEW

**Comment by remibergsma** [comment]
go build

**Comment by remibergsma** [comment]
go build

**Comment by eliasgomes** [comment]
go build

**Comment by eliasgomes** [comment]
go build

**Comment by remibergsma** [comment]
Upgrade tested:

```
2017-01-12 17:37:01.010 DEBUG [c.c.u.d.ScriptRunner] -- Schema upgrade from 5.2.0 to 5.3.0;
2017-01-12 17:37:01.010 DEBUG [c.c.u.d.ScriptRunner] --;
2017-01-12 17:37:01.010 DEBUG [c.c.u.d.ScriptRunner] UPDATE networks SET guest_type='Private' WHERE name LIKE 'vpc-%-privateNetwork' and guest_type='Isolated' 
2017-01-12 17:37:01.011 INFO  [c.c.u.d.Upgrade520to530] Performing data migration from 5.2.0 to 5.3.0



2017-01-12 17:37:01.324 DEBUG [c.c.u.d.ScriptRunner] --;
2017-01-12 17:37:01.324 DEBUG [c.c.u.d.ScriptRunner] -- Schema cleanup from 5.2.0 to 5.3.0;
2017-01-12 17:37:01.324 DEBUG [c.c.u.d.ScriptRunner] --;
2017-01-12 17:37:01.324 DEBUG [c.c.u.d.ScriptRunner] -- Backup table for later reference
2017-01-12 17:37:01.324 DEBUG [c.c.u.d.ScriptRunner] CREATE TABLE vpc_gateways_pre530 LIKE vpc_gateways 
2017-01-12 17:37:01.325 DEBUG [c.c.u.d.ScriptRunner] INSERT vpc_gateways_pre530 SELECT * FROM vpc_gateways 
2017-01-12 17:37:01.326 DEBUG [c.c.u.d.ScriptRunner] -- delete from original
2017-01-12 17:37:01.326 DEBUG [c.c.u.d.ScriptRunner] ALTER TABLE `vpc_gateways` DROP COLUMN `gateway` 
2017-01-12 17:37:01.328 DEBUG [c.c.u.d.ScriptRunner] ALTER TABLE `vpc_gateways` DROP COLUMN `netmask` 
2017-01-12 17:37:01.331 DEBUG [c.c.u.d.ScriptRunner] ALTER TABLE `vpc_gateways` DROP COLUMN `vlan_tag` 
2017-01-12 17:37:01.333 DEBUG [c.c.u.DatabaseUpgradeChecker] Cleanup script /var/lib/tomcat/webapps/client/WEB-INF/classes/db/schema-520to530-cleanup.sql is executed successfully
2017-01-12 17:37:01.334 DEBUG [c.c.u.DatabaseUpgradeChecker] Upgrade completed for version 5.3.0
```

**Review by borisroman — APPROVED** [review]
